src/apps/lwaftr/lwaftr.lua

```diff
@@ -307,13 +307,30 @@ local function cannot_fragment_df_packet_error(lwstate, pkt)
                                  ethernet_header_size, icmp_config)
 end
 
--- Given a packet containing IPv4 and Ethernet, encapsulate the IPv4 portion.
-local function ipv6_encapsulate(lwstate, pkt, next_hdr_type, ipv6_src, ipv6_dst,
-                                 ether_src, ether_dst)
-   -- TODO: decrement the IPv4 ttl as this is part of forwarding
-   -- TODO: do not encapsulate if ttl was already 0; send icmp
+local function encapsulate_and_transmit(lwstate, pkt, ipv6_dst, ipv6_src)
+   -- Do not encapsulate packets that now have a ttl of zero or wrapped around
+   local ttl = decrement_ttl(pkt)
+   if ttl == 0 or ttl == 255 then
```

Comment:
We're checking against both as a safeguard. Obviously wrap-around shouldn't happen, but that's there because it's cheap and makes the code robust against other changes.